src/lib.rs

```diff
@@ -520,13 +520,15 @@ fn json_arr_pop(ctx: &Context, args: Vec<String>) -> RedisResult {
     let key = ctx.open_key_writable(&key);
     let mut res = Value::Null;
 
-    key.get_value::<RedisJSON>(&REDIS_JSON_TYPE)?
+    let _: RedisResult = key
```

Comment:
It seems like you're ignoring any error that could occur (such as the `nonexistent_key`), since the result is thrown away with the `let _`.

Instead, I'd suggest the following code. Note the question mark at the end to handle the error. 
It also gets rid of the unnecessary mapping of the result value (which is unused) and therefore also doesn't require the explicit type annotation:

```
    key.get_value::<RedisJSON>(&REDIS_JSON_TYPE)?
        .ok_or_else(RedisError::nonexistent_key)
        .and_then(|doc| {
            doc.value_op(&path, |value| do_json_arr_pop(index, &mut res, value))
                .map_err(|e| e.into())
        })?;
```

Reply:
fix